pkg/intercp/catalog/heartbeat_component.go

```diff
@@ -0,0 +1,127 @@
+package catalog
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/api/system/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/core/user"
+)
+
+var heartbeatLog = core.Log.WithName("intercp").WithName("catalog").WithName("heartbeat")
+
+type heartbeatComponent struct {
+	catalog     Catalog
+	newClientFn NewClientFn
+	request     *v1alpha1.PingRequest
+	interval    time.Duration
+
+	leader *Instance
+	client Client
+}
+
+var _ component.Component = &heartbeatComponent{}
+
+type NewClientFn = func(url string) (Client, error)
+
+func NewHeartbeatComponent(
+	catalog Catalog,
+	instance Instance,
+	interval time.Duration,
+	newClientFn NewClientFn,
+) component.Component {
+	return &heartbeatComponent{
+		catalog: catalog,
+		request: &v1alpha1.PingRequest{
+			InstanceId:  instance.Id,
+			Address:     instance.Address,
+			InterCpPort: instance.InterCpPort,
+		},
+		newClientFn: newClientFn,
+		interval:    interval,
+	}
+}
+
+func (h *heartbeatComponent) Start(stop <-chan struct{}) error {
+	heartbeatLog.Info("starting heartbeats to a leader")
+	ctx := user.Ctx(context.Background(), user.ControlPlane)
```

Comment:
we have access to the context when creating this component, maybe we should pass it in `NewHeartbeatComponent` and then reuse here?

Reply:
While I could pass this in `New...`, I don't think this is a good idea. The ideal solution here would be that `Start` receives the context which is already enriched with the user information.
I'd rather follow the pattern of other components where every component starts it's own background context.

I can't find an issue to change chan to context, I can create it if needed.